### PR TITLE
Instructor: enroll students: blank lines processed by back-end during enrollment #9071

### LINKED ISSUE #9071: Instructor: enroll students: blank lines processed by back-end during enrollment
**Environment**: 
`master branch`

**Steps to reproduce**
1. Copy rows of student entries from external sources e.g. Microsoft Excel. Sample data shown below:

image

2. Paste the content into the `New Students` spreadsheet using `Ctrl + V` or `Cmd + V`:

image

3. Click the `Enroll students` button.

**Expected behaviour**
1. Success enrollment. Example shown below:

image

**Actual behaviour**
1. Blank lines processed by back-end during enrollment:

image

**Additional info**
1. This is due to potential line feeds/carriage returns during the process of copying data from external sources.
2. This causes the backend method `getRequestParamValue` to read in the line feeds/carriage returns, resulting in student entries to be read separated rows. Example shown below:

image

### PR DESCRIPTION
**Pull request by tanhengyeow**
Fixes #9071

1. Hotfix to address issue faced.

### CODE REVIEW

File: src/main/webapp/dev/js/common/instructorEnroll.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ function getUpdatedHeaderString(handsontableColHeader) {
 function getUserDataRows(spreadsheetData) {
     // needs to check for '' as an initial empty row with null values will be converted to e.g. "||||" after .map
     return spreadsheetData.filter(row => (!row.every(cell => cell === null || cell === '')))
-            .map(row => row.join('|'))
+            .map(row => row.join('|').replace(/\n|\r/g, ''))
```
Comment (xpdavid) [inline src/main/webapp/dev/js/common/instructorEnroll.js]: For every method in the chain, shouldn't we start a new line?
```
.map()
.replace()
.join()
```

In addition, do we have test to cover the cases?

Comment (tanhengyeow) [reply src/main/webapp/dev/js/common/instructorEnroll.js]: > For every method in the chain, shouldn't we start a new line?

@xpdavid The current way works but yes I agree the suggested way is easier to read. Will make the change :)

> In addition, do we have test to cover the cases?

@damithc suggested to release a hotfix first so right now I haven't added tests for this case. However I did experimented a bit and adding tests might be a little tricky. 

As of now, I can only replicate this issue by copying data from `.xlsx` files and pasting it in the `New Students` spreadsheet. Copying and pasting from Google Sheets would not cause this issue.

**Comment by xpdavid** [comment]
Oh. @tanhengyeow And you should submit the PR against release branch as it is hotfix.

**Comment by tanhengyeow** [comment]
Closing this as a separate PR is made to the `release` branch.

**Pull request closed without merging**